### PR TITLE
[CIVP-24102] upgrade pyyaml to 5.2

### PR DESCRIPTION
**Pull request by kmshelley**
Updating pyyaml since a client needs a more recent version

console pr to update default tag:
https://github.com/civisanalytics/console/pull/15808

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ dependencies:
 - pycrypto=2.6.1
 - pytest=5.3.5
 - python=3.7.6
-- pyyaml=3.13
+- pyyaml=5.2
```
Comment (jacksonlee-civis) [inline environment.yml:43]: Out of curiosity, is there a reason why the latest 5.4.1 (https://pypi.org/project/PyYAML/5.4.1/) cannot be used? For example, would it conflict with other dependencies in the image (and therefore crash the CircleCI build)?

**Review by jacksonlee-civis — COMMENTED** [review]
I have a question about the PyYAML version, in the comment below.

Also, apart from the default ds-py image tag for platform container scripts, there are other places I can think of that would probably need updating:

* The [ds-py image tag in the civis-jupyter-python3 image](https://github.com/civisanalytics/civis-jupyter-python3/blob/bb018437368e45d2c60501b04f6c163537b9555a/Dockerfile#L1), plus a new release of this image
* Relatedly, the default tag of civis-jupyter-python3 in the console repo, for the whatever the new tag is from the previous bullet point

**Comment by kmshelley** [comment]
Hi Jackson, when I tried to build off of pyyaml 5.4.1 I got nasty errors before the build failed. Here's a snippet of the error:
```
Examining conflict for notebook scikit-learn scipy pip pycrypto s3fs boto3 jinja2 patsy beautifulsoup4 jsonschema pycparser idna numpy psycopg2 brotlipy glmnet bqplot nose certifi tqdm dask seaborn setuptools cryptography ipython joblib pysocks jupyter conda cffi six pyarrow ruamel_yaml matplotlib click pyopenssl ipywidgets cloudpickle wheel pytest urllib3 requests conda-package-handling chardet numexpr p
```

**Review by jacksonlee-civis — APPROVED** [review]
LGTM

**Pull request merged**